### DISCUSSION TITLE
Are no-cache, no-store and revalidate 0 all effectively the same?

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#50117] [App Router] [answered]
Is there a difference when using `no-store`, `no-cache`, or `revalidate = 0`?

**Comment by ciruz** [comment, marked as answer]
No, all the same.

> **Requests are not cached if:**
> `revalidate: 0` or `cache: 'no-store'` is configured on individual fetch

See: https://nextjs.org/docs/app/building-your-application/data-fetching/caching#fetch

> **Good to know:**
> The `no-cache` option behaves the same way as `no-store` in Next.js.

See: https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache

**Reply by codinginflow** [reply]
Thank you for the clarification!

**Comment by joepetrillo** [comment]
I feel like these should be combined into one.
